### ISSUE TITLE
Case sensitivity when creating commit

### ISSUE THREAD

**Issue by justindbaur** [tinglesoftware/dependabot-azure-devops#138]
## Problem
Hi, I want to preface this with saying I don't think this is a bug on the side of this software even though the end of this will contain a couple of suggestions that if you feel like taking I would be more than willing to contribute them. I am *mostly* writing this issue as a way to help other people in case they ever find themselves in my situation. Onto the issue.

I found myself getting this issue: `Pull Request creation failed with status 400. Message: TF401398: The pull request cannot be activated because the source and/or the target branch no longer exists, or the requested refs are not branches (StandardError)`.  This was on my second repository I was setting up with the dependabot extension so deep down I didn't think it had anything to do with security (not that it stopped my from messing with it for several hours). I found myself reading through issue #40 which helped in me thinking it was security and somehow a different user running this pipeline than the other one.  That might be the problem for others but it wasn't for me. So after playing with security and eventually adding in my own PAT token but still getting the error I started getting desperate. I started reading both [update-script.rb](https://github.com/tinglesoftware/dependabot-azure-devops/blob/main/src/script/update-script.rb) and the [dependabot/dependabot-core](https://github.com/dependabot/dependabot-core) repo to work out what was going on and where my *real* issue might be originating from. 

## Investigation
So after reading the code and tracing what I was pretty sure was happening behind the scenes I had pretty much set my target on [this](https://github.com/dependabot/dependabot-core/blob/main/common/lib/dependabot/clients/azure.rb#L138) method where a commit was being created on another branch. I noticed that where it [gets called](https://github.com/dependabot/dependabot-core/blob/main/common/lib/dependabot/pull_request_creator/azure.rb#L29) from and I saw that there wasn't really any validation on whether or not it worked. That sorta sealed it for me that that call was failing to create the commit/branch and then it was trying to create a PR against it and that is where the issue showed itself. But I still didn't know *why* that call was failing and this is where I got a little inventive. I am not a ruby person so maybe this is elementary to better people than me but I noticed both your code and the dependabot-core code uses [`Excon`](https://github.com/excon/excon) to make the HTTP calls so I was hoping there would be some sort of way to view both the requests and responses made using it.  They allow you to set an environment variable  [`EXCON_DEBUG=1`](https://github.com/excon/excon/blob/master/README.md#options) and it will do some very verbose logging of the entire call. 

At this point I comment out my dependabot task in my yml file and switch to running `docker run` directly and add `-e EXCON_DEBUG=1` at the end of all the other `-e` arguments. This finally showed me the error I was suspecting I was getting from the `create_commit` method. The message of the response was `The path '/path1/path2/path3/MyProject.csproj' does not exist at commit '[CommitHash]'\\r\\nParameter name: newPush`. That path looked totally fine to me but after a while I did notice that the path in my repository was slightly different.  

```diff
- /path1/path2/path3/MyProject.csproj
+ /path1/path2/Path3/MyProject.csproj
```

## Solution
So I mocked up that commit with postman against `https://dev.azure.com/{organization}/{project}/_apis/git/repositories/{repository}/pushes` and I changed the `p` to `P` in Path3 and the commit went through! Awesome, but why is dependabot trying to use. So I decided to search my entire repository for that path and lo and behold it was in my `.sln` file. My solution worked just as expected even with the different casing but the azure devops api is not so kind.  So I switched the casing in my `.sln` file and boom dependabot worked. I then switch back to the task from the manual `docker run` and it still worked just fine. 

TL;DR Azure DevOps is case sensitive in it's creating of a commit but Visual Studio is not so you might not notice that they even differ. 

## Proposal
I think it would be super useful to be able to either set miscellaneous environment variables that would trickle their way into the docker container so that someone could be something like an environment variable that can be optionally set akin to `DEPENDABOT_EXTRA_CREDENTIALS` or it could be another input like this. 

```yml
- task: dependabot@1
  inputs: 
    ...
    extraEnvironmentVariables: EXCON_DEBUG=1
```

It could be a delimited list that could be read in and added to the docker args. 

The third option could be to tap into the `System.Debug` variable and when that is set you can add the `EXCON_DEBUG=1` variable. This one is my personal favorite.

**Comment by mburumaxwell** [comment]
@justindbaur thanks for spending the time to explain this issue and to even give the solution. Indeed, the passing of env from the task inputs is a good idea and a solution to quick debug. A PR is welcome anytime.
